_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,644 @@
+
+### Run the Kubernetes sub-generator
+
+Now let's generate the Kubernetes descriptors using JHipster. During the process, the generator will prompt for an FQDN (Fully qualified domain name) for the Ingress services, so let's first create a public IP on Google Cloud. With the help of [nip.io](nip.io), if you set <public-ip>.nip.io as your FQDN, you can test the deployment without having to purchase a real domain.
+
+Google Cloud provides a [free tier](https://cloud.google.com/free) of their services that grants you $300 in free credits if you are a new user.
+
+After you sign up, install [`gcloud` CLI](https://cloud.google.com/sdk/docs/install). When you reach the end of the process, the last step is to run `gcloud init` and set up authorization for the tool. Also, install the [Kubectl authentication plugin](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke).
+
+```shell
+gcloud compute addresses create gateway-ip --global
+```
+**Note**: The name for public IP must be the gateway or monolith application base name, you can find it in the JDL. For the current example, the application base name is `gateway`.
+
+```
+application {
+  config {
+    baseName gateway
+    applicationType gateway
+```
+
+Find out the IP address with the following command:
+
+```shell
+gcloud compute addresses describe gateway-ip --global --format='value(address)'
+```
+
+Create a `kuberntes` folder at the root of the project, and run the generator:
+
+```shell
+jhipster kubernetes
+```
+
+Choose the following options when prompted:
+
+- Type of application: **Microservice application**
+- Root directory: **../**
+- Which applications? (select all)
+- Set up monitoring? **No**
+- Which applications with clustered databases? select **store**
+- Kubernetes namespace: **demo**
+- Docker repository name: **your-dockerhub-username**
+- Command to push Docker image: `docker push`
+- Enable Istio? **No**
+- Kubernetes service type? **Ingress**
+- Kubernetes ingress type? **Google Kubernetes Engine Ingress**
+- Root FQDN for ingress services: **\<public-ip\>.nip.io**
+- Use dynamic storage provisioning? **Yes**
+- Use a specific storage class? (leave empty)
+
+### Customize the configuration
+
+Let's Encrypt requires an email address to remind you to renew the certificate after 30 days before expiry. You will only receive this email if something goes wrong when renewing the certificate with cert-manager. Set the email address in the issuer yaml descriptor `kubernetes/cert-manager/letsencrypt-staging-issuer.yml`:
+
+```yml
+spec:
+  acme:
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
+    email: <your-email>
+```
+
+### Deploy to Google Kubernetes Engine
+
+Create a Kubernetes cluster on Google Cloud:
+
+```shell
+gcloud container clusters create jhipster-cluster \
+--zone southamerica-east1-a \
```

Comment:
Is southamerica used for any particular reason? I remember Andrew having to create something in Australia once because it wasn't available elsewhere.

Reply:
Nevermind,  you answered this question below.